typespec-extension/src/code-model-builder.ts

```diff
@@ -904,7 +903,11 @@ export class CodeModelBuilder {
 
   private processParameter(op: CodeModelOperation, param: HttpOperationParameter, clientContext: ClientContext) {
     function isSubscriptionId(sdkContext: SdkContext, param: HttpOperationParameter): boolean {
-      return "subscriptionId".toLocaleLowerCase() === param?.name?.toLocaleLowerCase() && param.param && isArmCommonType(param.param);
+      return (
+        "subscriptionId".toLocaleLowerCase() === param?.name?.toLocaleLowerCase() &&
+        param.param &&
```

Comment:
nit, maybe also verify that it is path param.

Reply:
added